onert-micro/onert-micro/include/core/OMTensorData.h

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2025 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef ONERT_MICRO_CORE_TENSOR_DATA_H
+#define ONERT_MICRO_CORE_TENSOR_DATA_H
+
+#include "OMQuantizationData.h"
+
+#include <memory>
+#include <type_traits>
+
+namespace onert_micro
+{
+namespace core
+{
+
+// clang-format off
+
+// ------------------------------------------------------------------------------------------------
+
+template <typename T>
+class OMTensorData
+{
+  using QuantData = OMQuantizationData<T>;
+  using QuantDataPtr = std::unique_ptr<QuantData>;
+
+private:
+  T *_data = nullptr;
+  size_t _size = 0;
+  QuantDataPtr _quant = {};
```

Comment:
For `floats` QuantDataPtr will be redundant, but it's overhead will be small, i suppose.